platform/pulp/platform/models/repository.py

```diff
@@ -105,35 +105,44 @@ def natural_key(self):
         return (self.name,)
 
 
-class Plugin(MasterModel):
+class ContentAdaptor(MasterModel):
     """
-    An Abstract model for plugins.
+    An Abstract model for objects that import or publish content.
 
     Fields:
 
-    :cvar name: The plugin name.
+    :cvar name: The ContentAdaptor name.
     :type type: models.TextField
 
-    :cvar type: The plugin type.
+    :cvar type: The ContentAdaptor type.
     :type type: models.TextField
 
-    :cvar last_updated: When the plugin was last updated.
+    :cvar last_updated: When the adaptor was last updated.
     :type last_updated: fields.DateTimeField
 
     Relations:
 
     """
-    name = models.TextField(db_index=True)
+    name = models.TextField(db_index=True, unique=True)
     type = models.TextField(blank=False, default=None)
     last_updated = models.DateTimeField(auto_now=True, blank=True, null=True)
 
     class Meta:
         abstract = True
 
+    def natural_key(self):
```

Comment:
Will that be somewhat awkward? I'm failing to imagine example values for this field that will be useful _and_ unique, but I'm also not really sure what people have in mind for this.

Even if this is left as-is, I think the natural key on the Publisher should include the repository somehow, since there's no longer an intermediate table and the Publisher requires a repository relationship.